### ISSUE TITLE
Create Site Collection fails if Site Collection Admin has no email from people picker

### ISSUE THREAD

**Issue by gavinbarron** [OfficeDev/PnP-Partner-Pack#102]
Currently the code that creates the job entry assumes that the user entered for the primary and secondary site collection administrator has the email returned from the people picker.

Values for PrimarySiteCollectionAdmin and SecondarySiteCollectionAdmin sent to the controller:
```
PrimarySiteCollectionAdmin:[{"Login":"i:0#.f|membership|admin@[tenant].onmicrosoft.com","Name":"Admi Strato","Email":""}]
SecondarySiteCollectionAdmin:[{"Login":"i:0#.f|membership|admin@[tenant].onmicrosoft.com","Name":"Admi Strato","Email":""}]
```
The following code (lines 118 - 121 in HomeController.cs) 
```ob.PrimarySiteCollectionAdmin = model.PrimarySiteCollectionAdmin != null &&
                            model.PrimarySiteCollectionAdmin.Length > 0 ? model.PrimarySiteCollectionAdmin[0].Email : null;
                        job.SecondarySiteCollectionAdmin = model.SecondarySiteCollectionAdmin != null &&
                            model.SecondarySiteCollectionAdmin.Length > 0 ? model.SecondarySiteCollectionAdmin[0].Email : null;```

This results in the following job being persisted in the SharePoint list:
```
{
  "PrimarySiteCollectionAdmin": "",
  "SecondarySiteCollectionAdmin": "",
  "StorageMaximumLevel": 0,
  "StorageWarningLevel": 0,
  "UserCodeMaximumLevel": 0,
  "UserCodeWarningLevel": 0,
  "ExternalSharingEnabled": false,
  "PartnerPackExtensionsEnabled": true,
  "ResponsiveDesignEnabled": true,
  "SiteTitle": "asdfasdf",
  "RelativeUrl": "/sites/asdfasdf",
  "SitePolicy": "LBI",
  "Description": null,
  "Language": 1025,
  "ProvisioningTemplateUrl": "https://[tenant].sharepoint.com/sites/PnP-Partner-Pack-Infrastructure/PnPProvisioningTemplates/TestDriveRoot.xml",
  "TimeZone": 39,
  "TemplateParameters": null,
  "JobId": "7c4645e5-bd43-4074-a0ef-07f2a0a8c454",
  "Title": "Provisioning of Site Collection \"/sites/asdfasdf\" with Template \"https://[tenant].sharepoint.com/sites/PnP-Partner-Pack-Infrastructure/PnPProvisioningTemplates/TestDriveRoot.xml\" by admin@[tenant].onmicrosoft.com",
  "Owner": "admin@[tenant].onmicrosoft.com",
  "Status": 2,
  "ErrorMessage": null
}
```
When the job for this request is executed it fails with the message: **The Owner property is required for site creation and was not supplied.**

In this case the owner was the also the account selected using the people picker. 

Perhaps if both the Site Collection Admins are null this process should throw and exception before persisting the job as

**Comment by PaoloPia** [comment]
Hi,
Thanks for the feedback. We will update the behavior accordingly.

**Comment by 2P2-Anthony** [comment]
Paolo...

Do you have a patch for this bug yet? I'm hitting the same issue during POC testing in an environment that doesn't have Email accounts. 

If we manually update the code to use the login account instead, will that cause other issues?

**Comment by gavinbarron** [comment]
I patched my version to fall back to the job.Owner if the site collection
admin couldn't be resolved by changing the end of the CreateSiteCollection
method on the HomeController like this:

```

                        //Safety catch...
                        if
(string.IsNullOrEmpty(job.PrimarySiteCollectionAdmin))
                        {
                            job.PrimarySiteCollectionAdmin = job.Owner;
                        }


                        model.JobId =
ProvisioningRepositoryFactory.Current.EnqueueProvisioningJob(job);
```
It's a bit of a hack but it serves our purpose.

Paolo, if you want this hack as a 'fix' then I can submit a PR.

**Comment by 2P2-Anthony** [comment]
If the Job just needs an "email", why won't the [user]@[tenant].onmicrosoft.com work that they use for login?

**Comment by PaoloPia** [comment]
Hi Gavin,
Yes please. Feel free to submit a PR, I will look into it. Thanks.

**Comment by PaoloPia** [comment]
Fixed by merging #124 . I will close this one.
